### PR TITLE
Upgrade to MapboxMaps `v10.5.0`, MapboxNavigationNative `v99.0.0`

### PR DESCRIPTION
**Pull request by jill-cardamon**
PR updates dependencies to:
* Mapbox Maps v10.5.0.
* Mapbox Navigation Native v99.0.0.

### CODE REVIEW

**Comment by 1ec5** [comment]
If mapbox/mapbox-directions-swift#690 lands in time, we can add the name to the conversion from TollCollectionInfo to TollCollection:

https://github.com/mapbox/mapbox-navigation-ios/blob/e7eabfc2030a62ed6be91a6f3211ae87a871f7f0/Sources/MapboxCoreNavigation/TollCollection.swift#L7-L16

**Comment by 1ec5** [comment]
[Builds are failing](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/8074/workflows/16e78d3b-2ccd-4401-a345-b0994832d12d/jobs/69670) with the following compiler error:

```
/Users/distiller/project/Sources/MapboxCoreNavigation/RerouteControllerDelegate.swift:29:15: error: type 'RerouteErrorType' has no member 'wrongRequest'
        case .wrongRequest:
             ~^~~~~~~~~~~~
```

File: Package.swift [hunk 1]
```diff
@@ -24,8 +24,8 @@ let package = Package(
     dependencies: [
         .package(name: "MapboxDirections", url: "https://github.com/mapbox/mapbox-directions-swift.git", from: "2.4.0"),
         .package(name: "MapboxMobileEvents", url: "https://github.com/mapbox/mapbox-events-ios.git", from: "1.0.0"),
-        .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "98.0.0"),
-        .package(name: "MapboxMaps", url: "https://github.com/mapbox/mapbox-maps-ios.git", .exact("10.5.0-rc.1")),
+        .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "99.0.0"),
+        .package(name: "MapboxMaps", url: "https://github.com/mapbox/mapbox-maps-ios.git", .exact("10.5.0")),
```
Comment (1ec5) [inline Package.swift:28]: This should be a `from:` dependency, now that MapboxMaps is on a final release. If we don’t depend on any new APIs in v10.5.0 specifically, then roll the minimum requirement back to v10.4.1, which was introduced in #3806.

File: MapboxNavigation.podspec [hunk 2]
```diff
@@ -44,7 +44,7 @@ Pod::Spec.new do |s|
   s.module_name = "MapboxNavigation"
 
   s.dependency "MapboxCoreNavigation", "#{s.version.to_s}"
-  s.dependency "MapboxMaps", "10.5.0-rc.1"
+  s.dependency "MapboxMaps", "10.5.0"
```
Comment (1ec5) [inline MapboxNavigation.podspec:47]: `">= 10.4.1", "< 11.0.0"`

File: Package.swift (hunk 1 above)
Comment (jill-cardamon) [reply Package.swift:28]: Ah will fix this now

**Comment by jill-cardamon** [comment]
I noticed the compiler error. I'm fixing issues from the build, but I wanted to open this PR before tackling all issues.

File: Package.swift (hunk 1 above)
Comment (MaximAlien) [reply Package.swift:28]: When setting `MapView.location.options.puckType` to `nil` behavior would be different on `v10.4.1` and `v10.5.0`. If we keep depending on `v10.4.1` as a minimum version we'll have to add check for both versions. I think it'd be better to depend on `v10.5.0` from now on.

File: Sources/MapboxCoreNavigation/RerouteControllerDelegate.swift [hunk 3]
```diff
@@ -17,17 +17,17 @@ protocol ReroutingControllerDelegate: AnyObject {
 public enum ReroutingError: Error {
     /// Could not correctly process the reroute.
     case routeError
-    /// Could not compose correct request for rerouting.
-    case wrongRequest
+    /// Cause of reroute error is unknown.
+    case unknown
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RerouteControllerDelegate.swift:21]: `unknown` seems like it should be the first case. (Not that it particularly matters for an Error enumeration that isn’t CaseIterable.)

Renaming this enumeration case would be a backwards-incompatible change. We should keep the `wrongRequest` case around but alias it to another value. mapbox/mapbox-navigation-android#5784 aliases it to `unknown`, but the implementation in MapboxNavigationNative seems to have replaced every occurrence of `wrongRequest` with `invalidInput`. We don’t have an `invalidInput` case, but I think `routeError` would be analogous perhaps? It’d be OK to alias it to `unknown` for now but change it to `routeError` later.

Comment (jill-cardamon) [reply Sources/MapboxCoreNavigation/RerouteControllerDelegate.swift:21]: Yes. We were just looking at the failing tests and trying to get some additional info some circleci. We weren't going to keep this change as is.

**Comment by 1ec5** [comment]
> If mapbox/mapbox-directions-swift#690 lands in time, we can add the name to the conversion from TollCollectionInfo to TollCollection:

This is now available in MapboxDirections v2.5.0-beta.1.

File: Sources/MapboxCoreNavigation/RerouteControllerDelegate.swift (hunk 3 above)
Comment (1ec5) [reply Sources/MapboxCoreNavigation/RerouteControllerDelegate.swift:23]: > mapbox/mapbox-navigation-android#5784 aliases it to `unknown`, but the implementation in MapboxNavigationNative seems to have replaced every occurrence of `wrongRequest` with `invalidInput`.

Per @LukasPaczos:

> On the Nav SDK side we only handle the response errors, and only for custom routers, that's why the `RouterErrorType.UNKNOWN` is predominantly used.

**Review by 1ec5 — COMMENTED** [review]
Suggested changelog entries:

* During turn-by-turn navigation, incidents along the route are now refreshed periodically along with traffic congestion. ([#3869](https://github.com/mapbox/mapbox-navigation-ios/pull/3869))
* When the user passes a named toll collection point, the `TollCollection` object obtained through the `Notification.Name.electronicHorizonDidPassRoadObject` notification now has the `TollCollection.name` property set. ([#3869](https://github.com/mapbox/mapbox-navigation-ios/pull/3869))
* Fixed an issue where a failure to calculate a route offline could result in a successful result being passed to a `Directions.RouteCompletionHandler`. ([#3869](https://github.com/mapbox/mapbox-navigation-ios/pull/3869))

File: Tests/MapboxCoreNavigationTests/RoutesCoordinatorTests.swift [hunk 4]
```diff
@@ -46,22 +46,27 @@ private extension RoutesCoordinatorTests {
         }
         let encoder = JSONEncoder()
         encoder.userInfo[.options] = routeOptions
-        guard let routeData = try? encoder.encode(route.route),
+        guard let routeData = try? encoder.encode(route.response),
```
Comment (MaximAlien) [inline Tests/MapboxCoreNavigationTests/RoutesCoordinatorTests.swift:49]: 👍

**Review by MaximAlien — APPROVED** [review]

**Pull request merged**